pinot-common/src/main/java/org/apache/pinot/common/utils/config/TierConfigUtils.java

```diff
@@ -47,6 +51,27 @@ public static boolean shouldRelocateToTiers(TableConfig tableConfig) {
     return CollectionUtils.isNotEmpty(tableConfig.getTierConfigsList());
   }
 
+  public static String getDataDirFromTierConfig(String tableNameWithType, String tierName, TableConfig tableConfig) {
```

Comment:
(minor) We don't need to pass in table name because that can be read from the table config. Also suggest renaming the method:
```suggestion
  public static String getDataDirForTier(TableConfig tableConfig, String tierName) {
```